frontend/javascripts/oxalis/model/sagas/mesh_saga.ts

```diff
@@ -1082,33 +1139,105 @@ function* handleRemoveSegment(action: RemoveSegmentAction) {
   yield* put(removeMeshAction(action.layerName, action.segmentId));
 }
 
-function removeMesh(action: RemoveMeshAction, removeFromScene: boolean = true): void {
+function* removeMesh(action: RemoveMeshAction, removeFromScene: boolean = true): Saga<void> {
   const { layerName } = action;
   const segmentId = action.segmentId;
 
   if (removeFromScene) {
-    getSceneController().segmentMeshController.removeMeshById(segmentId, layerName);
+    getSceneController().segmentMeshController.removeMeshById(segmentId, layerName, null);
   }
 
-  removeMapForSegment(layerName, segmentId);
+  for (const addCoordString of Object.keys(adhocMeshesMapByLayer)) {
+    removeMapForSegment(layerName, segmentId, addCoordString);
+  }
 }
 
 function* handleMeshVisibilityChange(action: UpdateMeshVisibilityAction): Saga<void> {
-  const { id, visibility, layerName } = action;
+  const { id, visibility, layerName, additionalCoordinates } = action;
   const { segmentMeshController } = yield* call(getSceneController);
-  segmentMeshController.setMeshVisibility(id, visibility, layerName);
+  segmentMeshController.setMeshVisibility(id, visibility, layerName, additionalCoordinates);
+}
+
+export function* handleAdditionalCoordinateUpdate(): Saga<void> {
+  yield* take("WK_READY");
+
+  let previousAdditionalCoordinates = yield* select((state) => state.flycam.additionalCoordinates);
+
+  while (true) {
+    const action = (yield* take(["SET_ADDITIONAL_COORDINATES"]) as any) as FlycamAction;
+    //satisfy TS
+    if (action.type === "SET_ADDITIONAL_COORDINATES") {
+      const { segmentMeshController } = yield* call(getSceneController);
+      const meshRecords = yield* call({
+        context: segmentMeshController,
+        fn: () => segmentMeshController.meshesGroupsPerSegmentationId,
+      });
+
+      if (action.values == null || action.values.length === 0) break;
+      const newAdditionalCoordinates = getAdditionalCoordinatesAsString(action.values);
+
+      for (const additionalCoordinates of [action.values, previousAdditionalCoordinates]) {
+        const currentAdditionalCoordinatesAsString =
+          getAdditionalCoordinatesAsString(additionalCoordinates);
+        const shouldBeVisible = currentAdditionalCoordinatesAsString === newAdditionalCoordinates;
+        const recordsOfLayers = meshRecords[currentAdditionalCoordinatesAsString];
+        if (recordsOfLayers != null) {
```

Comment:
```suggestion
        const recordsOfLayers = meshRecords[currentAdditionalCoordinatesAsString] || {};
        // no if-anymore
        ...
```

due to the `|| {}` the for-loop below will simply execute zero times.